File: 226-invert-binary-tree/226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
  if (!root) return root;
  
  let tempNode;
  tempNode = root.right;
  root.right = root.left;
  root.left = tempNode;
  
  if (root.left) {
    invertTree(root.left);
  }
  if (root.right) {
    invertTree(root.right);
  }  

  return root;
};

File: 226-invert-binary-tree/226-invert-binary-tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} root
# @return {TreeNode}
def invert_tree(root)
    return root unless root
    
    temp_node = root.left
    root.left = root.right
    root.right = temp_node
    
    invert_tree(root.left) if root.left
    invert_tree(root.right) if root.right
    
    root
end